src/Compiler/NodeVisitor/AddFormClassesNodeVisitor.php

```diff
@@ -64,15 +70,31 @@ public function enterNode(Node $node): ?Node
             }
             $dimFetch = $assign->expr;
 
+            if (!$this->isUiControl($dimFetch->var)) {
```

Comment:
Why not just check if it is formStack? It can be anything else, why checking uiControl?

Reply:
in this case we are looking for `$form = $this->formsStack[] = $this->global->uiControl[''formName']` and we are on `uiControl['formName']` node